js/components/common/Disclaimer.js

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+
+/**
+ * A React component representing the disclaimer popup
+ */
+export default class Disclaimer extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            dontShowDisclaimer: localStorage.getItem('DontShowDisclaimer')
+        }
+    }
+
+    handleClose = () => {
+        let disclaimer = document.getElementById("disclaimerDiv");
+        disclaimer.classList.toggle("dismiss");
+    }
+
+    handleCheck = () => {
```

Comment:
The checkbox can be both selected and deselected. So you need to access the event object to see whether the box is checked or unchecked, as this determines what to set the local storage value to.